kong/runloop/handler.lua

```diff
@@ -795,7 +795,6 @@ do
 
     local reinitialize_cache = cache_size ~= router_cache_size
     if reinitialize_cache then
-      router_cache:flush_all()
```

Comment:
@dndx do you think just setting cache to new value and let the GC do the job is good enough? Or am I missing how this is related?

Reply:
Yes, even if we `flush` the cache, the actual value held by the table will still need to be GC'ed later, so I doubt this will free up any more memory for us immediately.

Reply:
Ok, If that doesn't give any advantages (e.g. freeing memory earlier), then it is fine.

Reply:
@dndx should we still just `flush` the `router_cache` after router has been build. My main worry is that some routes get cached, that incremental is in meanwhile removing.